### DISCUSSION TITLE
x-model.fill not working on multiple checkboxes bound to array

### DISCUSSION THREAD

**Discussion by fernandcf** [alpinejs/alpine#3675] [7. Bugs (archived)] [answered]
[x-model.fill](https://alpinejs.dev/directives/model#fill) not working on multiple checkboxes bound to array

&lt;div x-data=&quot;{colors:[]}&quot;&gt;
    &lt;input type=&quot;checkbox&quot; value=&quot;red&quot; x-model.fill=&quot;colors&quot; checked&gt;
    &lt;input type=&quot;checkbox&quot; value=&quot;orange&quot; x-model.fill=&quot;colors&quot; checked&gt;
    &lt;input type=&quot;checkbox&quot; value=&quot;yellow&quot; x-model.fill=&quot;colors&quot;&gt;
  Colors:&lt;span x-text=&quot;colors&quot;&gt;&lt;/span&gt;
&lt;/div&gt;

**Comment by ekwoka** [comment, marked as answer]
Fix PR'd in #3676
